integration_tests/src/main/python/orc_write_test.py

```diff
@@ -95,6 +95,7 @@ def test_part_write_round_trip(spark_tmp_path, orc_gen):
 @ignore_order(local=True)
 @pytest.mark.parametrize('orc_gen', [int_gen], ids=idfn)
 @pytest.mark.parametrize('orc_impl', ["native", "hive"])
+@pytest.mark.skipif(is_spark_321cdh(), reason="https://github.com/NVIDIA/spark-rapids/issues/7403")
```

Comment:
PR says it closes this issue, and we normally don't point to closed issues in a skip reason.  Seems like this PR should not say it is closing the issue, and we should leave it open to track the problem?  If so this should be an xfail not a skip, as we expect this should pass at some point.  Skip is appropriate for tests that we don't ever expect to pass, even with future fixes.

Reply:
I discussed with @jlowe and he suggested to go the skip route if the CPU is failing with this error. I confirmed this is the case, the error happens on `with_cpu_session` when we invoke the `write` on the dataframe. So this specific to the CDH 3.2.1 distribution before we kick in.

I am going to change the `reason` in the test to this explanation and so we can move to close the issue.

Reply:
@jlowe pls see https://github.com/NVIDIA/spark-rapids/pull/7422/commits/2cc08d0e93125f79de7797fa50f9c834b873873b.